workspaces/src/rpc/patch.rs

```diff
@@ -86,7 +95,7 @@ impl<'a, 'b> ImportContractBuilder<'a, 'b> {
         if self.import_data {
             records.extend(
                 self.from_network
-                    .view_state_raw(self.account_id.clone(), None, None)
+                    .view_state_raw(self.account_id.clone(), None, self.block_id.clone())
```

Comment:
very optional, but can avoid the clone here and simplify the code very slightly:

```
diff --git a/workspaces/src/rpc/patch.rs b/workspaces/src/rpc/patch.rs
index a459fec..96cc5c2 100644
--- a/workspaces/src/rpc/patch.rs
+++ b/workspaces/src/rpc/patch.rs
@@ -57,13 +57,14 @@ impl<'a, 'b> ImportContractBuilder<'a, 'b> {
     }
 
     pub async fn transact(self) -> anyhow::Result<Contract> {
+        let account_id = self.account_id;
         let sk = SecretKey::from_seed(KeyType::ED25519, DEV_ACCOUNT_SEED);
         let pk = sk.public_key();
-        let signer = InMemorySigner::from_secret_key(self.account_id.clone(), sk);
+        let signer = InMemorySigner::from_secret_key(account_id.clone(), sk);
 
         let mut account_view = self
             .from_network
-            .view_account(self.account_id.clone(), self.block_id.clone())
+            .view_account(account_id.clone(), self.block_id.clone())
             .await?;
         if let Some(initial_balance) = self.initial_balance {
             account_view.amount = initial_balance;
@@ -71,11 +72,11 @@ impl<'a, 'b> ImportContractBuilder<'a, 'b> {
 
         let mut records = vec![
             StateRecord::Account {
-                account_id: self.account_id.clone(),
+                account_id: account_id.clone(),
                 account: account_view.clone().into(),
             },
             StateRecord::AccessKey {
-                account_id: self.account_id.clone(),
+                account_id: account_id.clone(),
                 public_key: pk.clone().into(),
                 access_key: AccessKey::full_access(),
             },
@@ -84,10 +85,10 @@ impl<'a, 'b> ImportContractBuilder<'a, 'b> {
         if account_view.code_hash != CryptoHash::default() {
             let code_view = self
                 .from_network
-                .view_code(self.account_id.clone(), self.block_id.clone())
+                .view_code(account_id.clone(), self.block_id.clone())
                 .await?;
             records.push(StateRecord::Contract {
-                account_id: self.account_id.clone(),
+                account_id: account_id.clone(),
                 code: code_view.code,
             });
         }
@@ -95,11 +96,11 @@ impl<'a, 'b> ImportContractBuilder<'a, 'b> {
         if self.import_data {
             records.extend(
                 self.from_network
-                    .view_state_raw(self.account_id.clone(), None, self.block_id.clone())
+                    .view_state_raw(account_id.clone(), None, self.block_id)
                     .await?
                     .into_iter()
                     .map(|(key, value)| StateRecord::Data {
-                        account_id: self.account_id.clone(),
+                        account_id: account_id.clone(),
                         data_key: key,
                         value,
                     }),
@@ -120,6 +121,6 @@ impl<'a, 'b> ImportContractBuilder<'a, 'b> {
             .await
             .map_err(|err| anyhow::anyhow!("Failed to patch state: {:?}", err))?;
 
-        Ok(Contract::new(self.account_id, signer))
+        Ok(Contract::new(account_id, signer))
     }
 }

```

Reply:
good eye, I'll try that out. Thanks!